### PR TITLE
[SM-949] Add Event Logs to Service Account

### PR DESCRIPTION
**Pull request by differsthecat**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Add Event Logs tab to Service Accounts
Relies on [this](https://github.com/bitwarden/server/pull/3336) server PR
## Code changes

- **bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/event-logs/service-account-event-log-api.service.ts:** Add service account event logs API service
- **bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/event-logs/service-accounts-events.component.html**: Add Component
- **bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/event-logs/service-accounts-events.component.ts:** Add Component that extends BaseEventsComponent
- **bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/service-account.component.html:** Add Event Logs tab link
- **bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/service-accounts-routing.module.ts:** Add Event logs route
- **bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/service-accounts.module.ts:** Add new component to the Service Accounts module

 ## Screenshots

https://github.com/bitwarden/clients/assets/8926729/418eff26-1cbd-4289-ab1b-85b0280b4a0b

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/event-logs/service-accounts-events.component.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      exportService,
+      platformUtilsService,
+      logService,
+      fileDownloadService
+    );
+  }
+
+  async ngOnInit() {
+    // eslint-disable-next-line rxjs-angular/prefer-takeuntil, rxjs/no-async-subscribe
+    this.route.parent.params.subscribe(async (params) => {
```
Comment (Thomas-Avery) [inline bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/event-logs/service-accounts-events.component.ts:46]:
```suggestion
    // eslint-disable-next-line rxjs/no-async-subscribe
    this.route.params.pipe(takeUntil(this.destroy$)).subscribe(async (params) => {
```

Do we want to use the `destroy$` here?

**Review by Thomas-Avery — CHANGES_REQUESTED** [review]
Looks great, thank you!

One minor thing to take a look at.

File: bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/event-logs/service-accounts-events.component.ts (hunk 1 above)
Comment (differsthecat) [reply bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/event-logs/service-accounts-events.component.ts:46]: Ah yes, let's use that 🙂

**Review by Thomas-Avery — APPROVED** [review]
LGTM!

**Review by coltonhurst — APPROVED** [review]
This looks great, thanks @differsthecat and @Thomas-Avery

**Pull request merged**